Src/Support/IntegrationTests/StorageApiTests.cs

```diff
@@ -108,24 +108,28 @@ public void ETags()
             {
                 // Set a label with If-None-Match header. This should not change any labels
                 bucket.Labels = new Dictionary<string, string> { ["label"] = "1" };
-                var req1 = client.Buckets.Patch(bucket, bucket.Name);
-                req1.ETagAction = Google.Apis.ETagAction.IfNoneMatch;
+                var req1 = client.Buckets.Patch(bucket, bucket.Name)
+                    .WithOptions(req => req.ETagAction = Google.Apis.ETagAction.IfNoneMatch);
+
                 var ex1 = Assert.Throws<GoogleApiException>(() => req1.Execute());
+
                 Assert.Contains("[Precondition Failed", ex1.Message);
                 // Check that labels have not been added, and etag is the same
                 var bucket1 = client.Buckets.Get(bucket.Name).Execute();
                 Assert.Empty(bucket1.Labels.EmptyIfNull());
                 Assert.Equal(bucket.ETag, bucket1.ETag);
                 // Set a label with If-Match header. This should change labels
-                var req2 = client.Buckets.Patch(bucket, bucket.Name);
-                req2.ETagAction = Google.Apis.ETagAction.IfMatch;
-                var bucket2 = req2.Execute();
+                var bucket2 = client.Buckets.Patch(bucket, bucket.Name)
+                    .WithOptions(req => req.ETagAction = Google.Apis.ETagAction.IfMatch)
```

Comment:
Maybe a better example of the usage of this would be to have:
`Action<StorageBaseRequest> MatchEtag => request => req.ETagAction = Google.Apis.ETagAction.IfMatch;
`
and then here and in the other tests:
`... client.Buckets.Patch(bucket, bucket.Name).WithOptions(MatchEtag);`

Reply:
I don't *think* so, because I don't imagine many bits of code actually make multiple requests with the same options. I think tests are a bit special there.
(This "demonstration" is mostly for the sake of the PR, btw. The samples repo is where I'd expect users to see this in action. We can take our cue from what works well there.)

Reply:
Yep agree.